pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -191,9 +192,38 @@ public void findEntryFailed(ManagedLedgerException exception, Optional<Position>
                 && (exception instanceof NonRecoverableLedgerException)) {
             log.warn("[{}][{}] read failed from ledger at position:{} : {}", topicName, subName, failedReadPosition,
                     exception.getMessage());
-            findEntryComplete(failedReadPosition.get(), ctx);
+            if (isLedgerNotExistException(((NonRecoverableLedgerException) exception).getBkErrorCode())) {
+                try {
+                    long failedLedgerId = failedReadPosition.get().getLedgerId();
+                    Position lastPositionInLedger = PositionImpl.get(failedLedgerId,
+                            cursor.getManagedLedger().getLedgerInfo(failedLedgerId).get().getEntries() - 1);
+                    log.info("[{}][{}] ledger not existed, will complete the last position of the non-existed"
+                                    + " ledger:{}", topicName, subName, lastPositionInLedger);
+                    findEntryComplete(lastPositionInLedger, ctx);
+                } catch (Exception e) {
```

Comment:
Any known exception goes here?

Reply:
Yes.  `getLedgerInfo(failedLedgerId).get()` at line199 will throw exception and we must catch it

Reply:
It's a fake async operation.
We can add a `getOptionalLedgerInfo(ledgerId)` in `ManagedLedger` to avoid this exception.

Reply:
Maybe we could use `ManagedLedgerImpl#getLedgersInfo`, which will also avoid this catch exception

Reply:
> We can add a getOptionalLedgerInfo(ledgerId) in ManagedLedger to avoid this exception

It makes sense to me.  Adding `getOptionalLedgerInfo` will make the code more  clear.  Updated. PTAL @Jason918